crates/miden-objects/src/account/component/template/storage/init_storage_data.rs

```diff
@@ -10,8 +12,12 @@ use super::StorageValueName;
 /// An [`InitStorageData`] can be created from a TOML string when the `std` feature flag is set.
 #[derive(Clone, Debug, Default)]
 pub struct InitStorageData {
+    // TODO: Both the below fields could be a single field with a two variant enum
+    // (eg, BTreeMap<StorageValueName, StorageTemplateValue>)
```

Comment:
Not too happy with how I had to refactor this struct and the constructor to account for the new changes but might be ok for now

Reply:
I agree a single entry would be nicer. I think that would implicitly guarantee that we don't specify an entry in each map for the same `StorageValueName`.